cachito/workers/scm.py

```diff
@@ -147,7 +167,23 @@ def fetch_source(self):
             ) as exc:
                 log.warning("Error handling archived artifact '%s': %s", previous_archive, exc)
 
-        self.clone_and_archive()
+        self.clone_and_archive(gitsubmodule)
+
+    def update_git_submodule(self, repo):
+        """Update git submodules.
+
+        For the given repo, update submodules. For Cachito request for `retrodep` repo and
+        `go-github` submodule, the dir structure would look like,
+        retrodep/go-github/<content_of_go-github_repo>
+
+        :param git.Repo repo: the repository object.
+        """
+        try:
+            log.debug(f"Git submodules for the requested repo are: {repo.submodules}")
+            repo.submodule_update()
+        except:  # noqa E722
```

Comment:
Can we capture specific exceptions here?

Reply:
None of the methods in the file has exception type defined. I'll add them all at once in a separate commit/PR

Reply:
I really think we should capture specific exceptions here. Handling errors like this could hide unrelated issues/bugs in our code.

From https://github.com/gitpython-developers/GitPython/blob/e4a83ff7910dc3617583da7e0965cd48a69bb669/git/objects/submodule/root.py, I see it raises at least `git.InvalidGitRepositoryError` and `ValueError`. However, it does capture all other exceptions that could be raised in a huge try/except block and raises.

If we are just catching any exception here, we should at least log the real exception that was raised instead of concealing it in a CachitoError. For instance, we could have sth like `except Exception as e` and log whatever we have in `e`.